tools/SchemaManager/Commands/ApplyCommand.cs

```diff
@@ -150,10 +150,17 @@ await Policy.Handle<SchemaManagerException>()
             }
         }
 
-        private static async Task UpgradeSchemaAsync(string connectionString, int version, string script, CancellationToken cancellationToken)
+        private static async Task UpgradeSchemaAsync(string connectionString, int version, string script, CancellationToken cancellationToken, bool isFullSchemaSnapshot = false)
         {
-            // check if the record for given version exists in failed status
-            await SchemaDataStore.DeleteSchemaVersionAsync(connectionString, version, SchemaDataStore.Failed, cancellationToken);
+            if (!isFullSchemaSnapshot || version > 5)
```

Comment:
>5 [](start = 51, length = 1)

Is there are place in code where you can have `static const int SchemaWithVersion ` = 5 ?Two different places in code rely on same constant, it's easy to make mistake, or lost track of them

Reply:
Good point! I don't think we have any common place for the tool and sqlServer to define static constants. This schema version '5' is added since we are changing the approach to insert schema version for full scripts for schema version>5. I don't think changing this later but we might plan to take this(version > 5) out when we update minSupportedVersion>5. So not sure,  if I should create a separate class just for this one to be shared. Thoughts?

Reply:
Can be totally wrong, but I would create static method in `SchemaInformation` class
something like:

```
///<summary>Checks is provided version already has schema version inside schema script.</summary>
public static bool ContainsVersionInSchema(int version) =>version>5;
```
and use it in both places.


---
In reply to: [509749340](https://github.com/microsoft/healthcare-shared-components/pull/117#discussion_r509749340) [](ancestors = 509749340)